### ISSUE TITLE
Group5 - Autopredictor

### ISSUE THREAD

**Issue by zgarciaj** [UBC-MDS/software-review-2024#16]
Submitting Author:  @s-voon 
All current maintainers: (@AnuBanga , @AReyH , @zgarciaj)
Package Name: autopredictor
One-Line Description of Package: Python package streamlines the process of selecting and assessing machine learning models, presenting a simplified approach for evaluating different regression models without intricate manual setup. 
Repository Link:  https://github.com/UBC-MDS/autopredictor
Version submitted:  [0.0.3](https://github.com/UBC-MDS/autopredictor/releases/tag/0.0.3) 
Editor: Tiffany Timbers  
Reviewer 1: Atabak Alishiri
Reviewer 2: Doris Cai  
Reviewer 3: Julia Everitt
Reviewer 4: Simon Frew
JOSS DOI: TBD
Version accepted: TBD 
Date accepted (month/day/year): TBD

---

## Code of Conduct & Commitment to Maintain Package

- [x] I agree to abide by [pyOpenSci's Code of Conduct][PyOpenSciCodeOfConduct] during the review process and in maintaining my package after should it be accepted.
- [x] I have read and will commit to package maintenance after the review as per the [pyOpenSci Policies Guidelines][Commitment].

## Description

- Include a brief paragraph describing what your package does:

## Scope

- Please indicate which category or categories. 
Check out our [package scope page][PackageCategories] to learn more about our 
scope. (If you are unsure of which category you fit, we suggest you make a pre-submission inquiry):

 - [x] Workflow automation
	- [ ] Data retrieval
	- [ ] Data extraction
	- [ ] Data processing/munging
	- [ ] Data deposition
	- [ ] Data validation and testing
	- [ ] Data visualization[^1]
	- [ ] Citation management and bibliometrics
	- [x] Scientific software wrappers
	- [ ] Database interoperability

Domain Specific & Community Partnerships 

	- [ ] Geospatial
	- [ ] Education
	- [ ] Pangeo
	

## Community Partnerships
If your package is associated with an 
existing community please check below:

- [ ] [Pangeo][pangeoWebsite]
	- [ ] My package adheres to the [Pangeo standards listed in the pyOpenSci peer review guidebook][PangeoCollaboration]

> [^1]: Please fill out a pre-submission inquiry before submitting a data visualization package.

- **For all submissions**, explain how the and why the package falls under the categories you indicated above. In your explanation, please address the following points (briefly, 1-2 sentences for each):  

  - Who is the target audience and what are scientific applications of this package?  

  - Are there other Python packages that accomplish the same thing? If so, how does yours differ?

  - If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or `@tag` the editor you contacted:

## Technical checks

For details about the pyOpenSci packaging requirements, see our [packaging guide][PackagingGuide]. Confirm each of the following by checking the box. This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] uses an [OSI approved license][OsiApprovedLicense].
- [x] contains a README with instructions for installing the development version. 
- [x] includes documentation with examples for all functions.
- [x] contains a tutorial with examples of its essential functions and uses.
- [x] has a test suite.
- [ ] has continuous integration setup, such as GitHub Actions CircleCI, and/or others.

## Publication Options

- [ ] Do you wish to automatically submit to the [Journal of Open Source Software][JournalOfOpenSourceSoftware]? If so:

<details>
 <summary>JOSS Checks</summary>  

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements][JossSubmissionRequirements]. Be aware that completing the pyOpenSci review process **does not** guarantee acceptance to JOSS. Be sure to read their submission requirements (linked above) if you are interested in submitting to JOSS.
- [ ] The package is not a "minor utility" as defined by JOSS's [submission requirements][JossSubmissionRequirements]: "Minor ‘utility’ packages, including ‘thin’ API clients, are not acceptable." pyOpenSci welcomes these packages under "Data Retrieval", but JOSS has slightly different criteria.
- [ ] The package contains a `paper.md` matching [JOSS's requirements][JossPaperRequirements] with a high-level description in the package root or in `inst/`.
- [ ] The package is deposited in a long-term repository with the DOI: 

*Note: JOSS accepts our review as theirs. You will NOT need to go through another full review. JOSS will only review your paper.md file. Be sure to link to this pyOpenSci issue when a JOSS issue is opened for your package. Also be sure to tell the JOSS editor that this is a pyOpenSci reviewed package once you reach this step.*
  
</details>

## Are you OK with Reviewers Submitting Issues and/or pull requests to your Repo Directly?
This option will allow reviewers to open smaller issues that can then be linked to PR's rather than submitting a more dense text based review. It will also allow you to demonstrate addressing the issue via PR links.

- [x] Yes I am OK with reviewers submitting requested changes as issues to my repo. Reviewers will then link to the issues in their submitted review.

Confirm each of the following by checking the box.

- [x] I have read the [author guide](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html). 
- [x] I expect to maintain this package for at least 2 years and can help find a replacement for the maintainer (team) if needed.

## Please fill out our survey

- [ ] [Last but not least please fill out our pre-review survey](https://forms.gle/F9mou7S3jhe8DMJ16). This helps us track
submission and improve our peer review process. We will also ask our reviewers 
and editors to fill this out.

**P.S.** Have feedback/comments about our review process? Leave a comment [here][Comments]

## Editor and Review Templates

The [editor template can be found here][Editor Template].

The [review template can be found here][Review Template].

[PackagingGuide]: https://www.pyopensci.org/python-package-guide/

[PackageCategories]: https://www.pyopensci.org/software-peer-review/about/package-scope.html

[JournalOfOpenSourceSoftware]: http://joss.theoj.org/

[JossSubmissionRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements

[JossPaperRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain

[PyOpenSciCodeOfConduct]: https://www.pyopensci.org/governance/CODE_OF_CONDUCT

[OsiApprovedLicense]: https://opensource.org/licenses

[Editor Template]: https://www.pyopensci.org/software-peer-review/appendices/templates.html#editor-s-template

[Review Template]: https://www.pyopensci.org/software-peer-review/appendices/templates.html#peer-review-template

[Comments]: https://pyopensci.discourse.group/

[PangeoCollaboration]: https://www.pyopensci.org/software-peer-review/partners/pangeo

[pangeoWebsite]: https://www.pangeo.io
[Commitment]: https://www.pyopensci.org/software-peer-review/our-process/policies.html#after-acceptance-package-ownership-and-maintenance

**Comment by dorisyycai** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [x] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [x] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [ ] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:**
  - [x] All tests pass on the reviewer's local machine for the package version submitted by the author. Ideally this should be a tagged version making it easy for reviewers to install.
  - [x] Tests cover essential functions of the package and a reasonable range of inputs and conditions.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [x] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [x] **A short summary** describing the high-level functionality of the software
- [x] **Authors:** A list of authors with their affiliations
- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [x] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing:

---

#### Review Comments
Overall, it is a comprehensive and useful package
The function tests achieved 100% test coverage, which is impressive
The vignette is well written, and the example chosen for demonstration is very helpful for understanding the project.
The use of badges in the README allows for easy navigation to various aspects of the package.

#### Recommendations:
It would be easier to read if the badges are organized in a table format, especially considering their increasing number.
It would be great to explicitly list the eight covered models on README, providing users with a quick understanding.
Include references or citations that can be referred to, to improve the package's transparency.
There's inconsistency on the CI/CD badge on README, showing a failure while the CI/CD action has actually passed.
Maybe include how to conduct unit tests and assess test coverage

**Comment by srfrew** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [ ] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [ ] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [x] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:**
  - [x] All tests pass on the reviewer's local machine for the package version submitted by the author. Ideally this should be a tagged version making it easy for reviewers to install.
  - [x] Tests cover essential functions of the package and a reasonable range of inputs and conditions.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS (NA)

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1

---

#### Review Comments
- The project vignette was clear and motivating! Further, the README and package documentation was well written and organized. This package is useful for EDA and selecting baseline model performance. Great work! Below are some recommendations to improve documentation and visuals, and one significant note on `display_best_score`.
- Installation instructions were overall clear! One note is that there were not instructions to include dev dependencies.
- I recommend that the models types to be trained are listed in the README and documentation of `fit`. It would likely be good to give a brief overview of the models themselves to orient the reader. 
- Does this project require a continuous response or can it handle categoricals? This may be worth noting. 
- I recommend that badges are organized into a table (visual preference)
- **Significant**: The function `display_best_score` returns the max value, which is incorrect for MAPE, MAE, MSE, and RMSE. It should be the minimum value in these cases, and maximum for R^2.

**Comment by atabak-alishiri** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [X] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [X] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [X] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [X] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [X] **Function Documentation:** for all user-facing functions.
- [X] **Examples** for all user-facing functions.
- [X] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [X] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [X] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [X] The package name
- [ ] Badges for:
    - [X] Continuous integration and test coverage,
    - [X] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [X] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [X] Short description of package goals.
- [X] Package installation instructions
- [X] Any additional setup required to use the package (authentication tokens, etc.)
- [ ] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [X] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [X] Link to your documentation website.
- [X] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [X] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [X] Package documentation is clear and easy to find and use.
- [X] The need for the package is clear
- [X] All functions have documentation and associated examples for use
- [X] The package is easy to install


#### Functionality

- [X] **Installation:** Installation succeeds as documented.
- [X] **Functionality:** Any functional claims of the software been confirmed.
- [X] **Performance:** Any performance claims of the software been confirmed.
- [X] **Automated tests:**
  - [X] All tests pass on the reviewer's local machine for the package version submitted by the author. Ideally this should be a tagged version making it easy for reviewers to install.
  - [X] Tests cover essential functions of the package and a reasonable range of inputs and conditions.
- [X] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [X] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [X] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [X] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1

---

#### Review Comments

The package is a robust tool designed to simplify predictive modeling by automating model selection and prediction tasks. Its comprehensive functionality and high test coverage are commendable, reflecting a strong commitment to quality and reliability. I really enjoyed reviewing the package.

**Strengths:**
- **Versatility:** Supports a wide range of models, making it suitable for diverse data science projects.
- **Documentation:** Well-written README and vignettes ensure ease of use, facilitating quick adoption by new users.
- **Test Coverage:** Impressive 100% test coverage guarantees stability and reduces the risk of bugs.

**Recommendations:**
- **Badge Organization:** Organizing badges in a table could enhance the README's readability and visual appeal.
- **Model Listings:** Clearly listing supported models in the README would immediately inform users about the package's capabilities.
- **References:** Including references or citations would enhance transparency and provide valuable insights into the methodologies implemented.
- **CI/CD Badge Accuracy:** Ensuring the CI/CD badge accurately reflects the current build status is essential for maintaining user trust.
- **Response Variable Types:** Specifying which types of response variables (continuous, categorical) the package can handle would clarify its applicability for potential users.
- **GitHub "About" Section:** Adding a brief description and links to the documents in the GitHub "About" area can make it easier for people to find and get to know what the package is for and how to use it.

In summary, the `autopredictor` package is a valuable asset in the data science toolkit, with its benefits significantly outweighing the minor areas for improvement. Addressing these recommendations could further enhance its usability and adoption. Well done fellas.

**Comment by juliaeveritt** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [ ] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [x] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [x] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:**
  - [x] All tests pass on the reviewer's local machine for the package version submitted by the author. Ideally this should be a tagged version making it easy for reviewers to install.
  - [x] Tests cover essential functions of the package and a reasonable range of inputs and conditions.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing:

1 hour

#### Review Comments

* ** ReadMe:** Documentation was easy to follow and clearly outlines the need for the package
* **Installation** Installation was easy to follow and all tests passed!
* ** ReadMe:** The custom logo is a nice touch!
* **ReadMe**: It would be useful to add more information about the 8 models in your documentation
* **ReadMe:** I would also find it helpful to see more specific criteria on input data format in your docs (ie explicitly defining clean data) 

Overall, really well done! Everything is well organized and easy to use, and my main critique is adding more information about the specifics of your functions and their outputs to your docs. :)
